cassiopeia/solver/ILPSolver.py

```diff
@@ -142,16 +150,25 @@ def solve(
 
         # determine diameter of the dataset by evaluating maximum distance to
         # the root from each sample
-        max_lca_distance = 0
-        lca_distances = [
-            dissimilarity_functions.hamming_distance(root, np.array(u))
-            for u in targets
-        ]
+        if (self.maximum_potential_graph_lca_height is not None) and (self.maximum_potential_graph_lca_height > 0):
+            max_lca_distance = self.maximum_potential_graph_lca_height
```

Comment:
No, I don't think it will because of the `else` clause right below. We only calculate the `maximum_potential_graph_lca_height` is not specified.

This also seems to work empirically when I run the code.